### ISSUE TITLE
Text and RenderTexture Issue

### ISSUE THREAD

**Issue by hugoruscitti** [photonstorm/phaser#5064]
## Version

* Phaser Version: 3.22.0

* Operating system: osx and windows.

* Browser: Chrome 80.0.3987.162 and Firefox 72.0.2 

## Description

I ran into a problem when I try to change a text after creating a texture. Text is displayed with edge flaws:

![Phaser3 - Text and RenderTexture Issue 2020-04-01 15-34-24](https://user-images.githubusercontent.com/99183/78176889-a0506980-7433-11ea-8893-bfdfa68c7137.png)

The problem only happens when I use `type: Phaser.AUTO` in the settings and after create a RenderTexture, and not if i use `type: Phaser.CANVAS`. It is not clear to me why it happens, I tried in various browsers and even osx and windows, on two different machines.

## Example Test Code

I wrote an example of this issue, its show a text (with no issues), then create a renderTexture and change the text to produce the text with edge flaws.

- https://codepen.io/hugoruscitti/pen/OJVGLEL

note that the issue only happens if the `var texture = this.add.renderTexture(0, 0, 200, 200);` line is present and you change the text after that texture creation.

The code show this text:

![Phaser3 - Text and RenderTexture Issue 2020-04-01 15-34-04](https://user-images.githubusercontent.com/99183/78178289-c545dc00-7435-11ea-89b3-ca72dfe5d9d0.png)

and a second later show the changed text with edge flaws:

![Phaser3 - Text and RenderTexture Issue 2020-04-01 15-34-24](https://user-images.githubusercontent.com/99183/78178301-caa32680-7435-11ea-8a65-06777ef6d696.png)

**Comment by samme** [comment]
`setVisible(false)` on the Render Texture doesn't prevent the problem, but reordering the objects does.

**Comment by esromneb** [comment]
I can second that I am getting this bug.  I made my own code pen but the one above shows the same effect.  https://codepen.io/esromneb/pen/WNQXmOv

**Comment by esromneb** [comment]
Seems like adding a new text object to the scene after the text has changed/broken will fix the issue.  This works until another render texture is added

**Comment by Kitsee** [comment]
This bug is caused by webGL's PMA texture parameter being set to false before updateCanvasTexture being called as updateCanvasTexture does not set the parameter back to the required value of true. This can be fixed by overridding the updateCanvasTexture of the renderer with the following
```
        this.game.renderer.updateCanvasTexture = function (srcCanvas, dstTexture, flipY)
        {
            if (flipY === undefined) { flipY = false; }
    
            var gl = this.gl;
    
            var width = srcCanvas.width;
            var height = srcCanvas.height;
    
            if (width > 0 && height > 0)
            {
                this.setTexture2D(dstTexture, 0);
                
                gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, flipY);
                gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, true);

                gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, srcCanvas);
                console.log("Hi");
                dstTexture.width = width;
                dstTexture.height = height;
        
                this.setTexture2D(null, 0);
            }
    
            return dstTexture;
        }
```

In our work-around's implementation we put this code in the init of a global plugin.

The following line of code is the fix that is missing in stock phaser. dont worry about it being hardcoded to true as it is hardcoded to true in its corresponding create method which sets the PMA correctly.
```
gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, true);
```

**Comment by Kitsee** [comment]
In our instance of this bug, spine sets this parameter to false causing the bug to appear.

**Comment by photonstorm** [comment]
Thank you for submitting this issue. We have fixed this and the fix has been pushed to the `master` branch. It will be part of the next release. If you get time to build and test it for yourself we would appreciate that.

**Comment by grauw** [comment]
Since we’re on Phaser 3.23.0, we have deployed the following local fix in our project:

```
// The `WebGLRenderer.updateCanvasTexture` method will now set `gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL`
// to true, which should stop issues where you update a Text Game Object, having added a Render
// Texture or Spine Game Object to the Scene after it, which switches the PMA setting. Fix #5064
// See https://github.com/photonstorm/phaser/issues/5064
// See https://github.com/photonstorm/phaser/commit/809b8a41d91
const WebGLRenderer = Phaser.Renderer.WebGL.WebGLRenderer;
WebGLRenderer.prototype._updateCanvasTexture = WebGLRenderer.prototype.updateCanvasTexture;
WebGLRenderer.prototype.updateCanvasTexture = function(srcCanvas, dstTexture, flipY) {
	if (srcCanvas.width > 0 && srcCanvas.height > 0) {
		this.gl.pixelStorei(this.gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, true);
	}
	return this._updateCanvasTexture(srcCanvas, dstTexture, flipY);
};
```

In case anyone runs into this issue prior to the fix’s release.
